code/Control/UserDefinedFormController.php

```diff
@@ -511,6 +528,152 @@ public function finished()
         ]);
     }
 
+    /**
+     * This returns a Confirm Folder form used to verify the upload folder for EditableFileFields
+     * @return ViewableData
+     */
+    public function confirmfolderform()
+    {
+        $request = $this->getRequest();
+        $id = $request->requestVar('ID');
+        $userFormID = $request->requestVar('UserFormID');
+
+        $editableFormField = EditableFormField::get()->byID($id);
+        if (!$editableFormField) {
+            $editableFormField = Versioned::get_by_stage(EditableFormField::class, Versioned::DRAFT)->byID($id);
+        }
+
+        $folderId = 0;
+        if (!$editableFormField) {
+            $editableFormField = new EditableFileField();
+        } elseif ($editableFormField instanceof EditableFileField) {
+            $folderId = $editableFormField->Folder->ID;
+        }
+
+
+        $fields = new FieldList();
+
+
+        $label = new LiteralField('Label', 'Files that your users upload should be stored carefully to reduce the risk of exposing sensitive data. Ensure the folder you select can only be viewed by appropriate parties. Folder permissions can be managed within the Files area.');
+        $label->addExtraClass(' mb-3');
+        $fields->push($label);
+
+
+        $userFormId = $request->requestVar('UserFormID');
+        $userForm = UserDefinedForm::get()->byID($userFormId);
+        $createFolder = new TextField('CreateFolder', '', $userForm->Title);
+        $createFolder->setDescription('<span class="font-icon-eye pr-1 align-middle" aria-hidden="true"></span>Restricted access, uploads will be visible to the following groups: Administrators');
+        $createFolder->addExtraClass('pt-2');
+        $fields->push($createFolder);
```

Comment:
Following on from my comment in the Entwine file, something along these lines should allow you to add the field prefix without hacking it in via jQuery:

```suggestion
        $fields->setSchemaData(['prefix' => 'SecureUploads']);
        $fields->push($createFolder);
```